### ISSUE TITLE
MongoDB access might be too slow

### ISSUE THREAD

**Issue by ryojp** [ryojp/todo#40]
On my Mac, login request takes around 1s to get valid token response, even though there are less than ten users registered in the DB, and the environment is set up locally using k8s.

This should require further investigation.

**Comment by ryojp** [comment]
I found that `bcrypt.compare()` is responsible for the most of the latency, not the DB access, when handling the login request.

I added `console.time()` and `console.timeEnd()` to measure each code section as below:
```
initial findOne: 2.364ms  // MongoDB access
getAuthenticated::comparePassword: 62.706ms  // bcrypt.compare()
getAuthenticated: 62.852ms
sign: 1.136ms  // jsonwebtoken.sign()
```

Additionally, I found that when the environment is set up using Docker Compose, it's much faster (~83ms), as opposed to Kubernetes with Istio injection (~300ms).
This is most likely because of the absence of reverse proxy (Istio IngressGateway) and sidecar proxies (Istio-injection).

In summary, the MongoDB access is not the critical path, closing the issue.
